xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -1180,10 +1181,11 @@ bool CDecoder::Open(AVCodecContext* avctx, AVCodecContext* mainctx, enum AVPixel
       // a driver may use multi-thread decoding internally
       m_refs += CServiceBroker::GetCPUInfo()->GetCPUCount();
     }
-    else
-      m_refs += CServiceBroker::GetCPUInfo()->GetCPUCount() / 2;
+
     // by specification hevc decoder can hold up to 8 unique refs
-    m_refs += avctx->refs ? avctx->refs : 8;
+    /* For some reason avctx->refs returns always 1 ref frame (tested
```

Comment:
It's not about whether the ffmpeg was the culprit. I don't approach it like that.

ffmpeg when it cannot know the number of refs frames returns 1

I have only changed the code from:
`m_refs += avctx->refs ? avctx->refs : 8;`

to:
`m_refs += (avctx->refs > 1) ? avctx->refs : 8;`

The previous code assumes that when it cannot be known it will return zero and it is not.

(*) I think there is some limitation in HEVC so that it is not possible to know the number of refs frames easily unless it is written as metadata in the header (as x265 does). You can always try to find out the exact reason why ffmpeg returns 1 but while there is nothing wrong with reserving space for 8 "possible" refs frames (max allowed for HEVC).